File: src/models/homepage.ts
import type { HomepageLink } from "./homepage-link";
import type { Partner } from "./partner";
import type { PartnerARD } from "./partner-ard";

export type Homepage = Readonly<{
  partnerARD?: PartnerARD
  // NOTE: Not sure if we can have more data about Turboself than just SSO.
  partnerTurboself?: Partner

  links: Array<HomepageLink>
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/Homepage.kt
package ink.literate.pawnote.models

data class Homepage(
    val partnerARD: PartnerARD? = null,
    // NOTE: Not sure if we can have more data about Turboself than just SSO.
    val partnerTurboself: Partner? = null,
    val links: List<HomepageLink>
)
